python/cudf/cudf/core/scalar.py

```diff
@@ -70,13 +114,24 @@ def __init__(self, value, dtype=None):
                 self._host_dtype = value._host_dtype
             else:
                 self._device_value = value._device_value
-        elif isinstance(value, cudf._lib.scalar.DeviceScalar):
-            self._device_value = value
         else:
             self._host_value, self._host_dtype = self._preprocess_host_value(
                 value, dtype
             )
 
+    @classmethod
+    def from_device_scalar(cls, dscalar):
+        if not isinstance(dscalar, cudf._lib.scalar.DeviceScalar):
+            raise TypeError(
+                "Expected an instance of DeviceScalar, "
+                f"got {type(dscalar).__name__}"
+            )
+        obj = object.__new__(cls)
+        obj._host_value = None
+        obj._host_dtype = None
```

Comment:
I'm not sure about the implications of doing this; I'd rather we tackled this in a future PR.

I don't think the switch to py38 was what was blocking this; we previously had a custom implementation of `cached_property`.

Reply:
This appears to be stale. It might have been from a time where `cudf.Scalar` did not cache and I was worried that inspecting the type would incur a sync (I don't think it actually _does_ in the end). This might have been part of what led me to want to just cache the entire value instead. Either way I am not sure any action is needed here. Caching the host dtype won't improve performance significantly - it should be able to obtain this information without incurring a sync no matter the state of the scalar AFAICT.